apps/iiif/manifests/models.py

```diff
@@ -353,3 +350,15 @@ class RelatedLink(models.Model):
     format = models.CharField(max_length=255, choices=Choices.MIMETYPES, blank=True, null=True)
     profile = models.CharField(max_length=255, blank=True, null=True)
     manifest = models.ForeignKey(Manifest, on_delete=models.CASCADE)
+
+@receiver(signals.pre_save, sender=Manifest)
+def set_logo_url(sender, instance, **kwargs):
+    """
+    Sets logo_url for manifests with logo file upload
+    """
+    # For logo image uploads (i.e. NOT remote manifest ingestion), either new or replacing an existing file
+    if instance.logo and instance.logo != None and instance.logo_url != f'{settings.HOSTNAME}/media/{instance.logo}':
+        instance.logo_url = f'{settings.HOSTNAME}/media/{instance.logo}'
+    # If logo upload file is cleared, logo_url should also be cleared
+    elif instance.logo == "":
+        instance.logo_url = ""
```

Comment:
Testing this out locally, every logo URL gets cleared from remote manifests because of this section. I would not worry about handling this automatically if the logo is cleared. Users can clear the logo URL manually if they are also manually clearing the logo file upload.

Reply:
Good point – removed the `elif` block